### ISSUE TITLE
Visible zones creating massive lag 

### ISSUE THREAD

**Issue by mloveless1227** [nikooo777/ckSurf#60]
The problem I'm having is when the zones are set to visible on any of the settings, it's causing massive fps issues for clients as well as large lag spikes for the server, where the SV on netgraph is spiking into the red. These spikes create nasty stutters every 10-15 seconds.

I have done a number of things to ensure the problem doesn't lay elsewhere. As soon as I set the visible zones cvar to 0, the problems stop. That's fine as a temporary solution. But it's really nice having visible start and end zones. I know this has been an outstanding issue for a while and was talked about on the allied mods forum. Hopefully it's not too much work to address it.

**Comment by 2called-chaos** [comment]
Yeah I feel you :) It's on my todo but honestly I don't quite even know what the issue is. One thing I figured so far is that it is either related to specific maps or the way zones are set up / colliding with stuff as I don't have this issue on every map, on some it's bad and on some it's just crashing my game.

I recorded a video a while back to illustrate the issue and I noticed a warning in the client (not even in the console) when playing in double the resolution (something client render overflowing something, can't read it in the video anymore)... https://www.youtube.com/watch?v=Vq2MsePN9HQ

**Comment by fluffyst** [comment]
Hey @2called-chaos, would you mind giving me the map name in the video? (Zone coords too would be great) May be able to help.

**Comment by 2called-chaos** [comment]
@fluffyst Hey there :) I'm not 100% sure if the zones are exactly the same but I can't remember changing them.

The map is bhop_bridge [[download from my fastdl]](https://bhgo.funcs.de/maps/funcs/bhop_bridge.bsp) and the zones 

| mapname | zoneid | zonetype | zonetypeid | pointa_x | pointa_y | pointa_z | pointb_x | pointb_y | pointb_z | vis | team | zonegroup | zonename |
| ------- | ------ | -------- | ---------- | -------- | -------- | -------- | -------- | -------- | -------- | --- | ---- | --------- | -------- |
| bhop_bridge | 0 | 1 | 0 | -2745.75 | 4617.64 | -420.969 | -2366.86 | 4365.49 | -560.969 | 0 | 0 | 0 | Start 0 |
| bhop_bridge | 1 | 2 | 0 | 12.4072 | -9492.26|237.031|-522.566|-9630.11|87.0312|0|0|0|Start 0|
| bhop_bridge | 2 | 1 | 0 | -305.086 | -9485.54|591.031|-208.269|-9380.47|501.031|0|0|1|BONUS |1|
| bhop_bridge | 3 | 2 | 0 | -325.216 | -9772.54|88.0312|-121.605|-9704.02|89.0312|0|0|1|BONUS 1|

What do you think it is? Since I'm coding on Mac and CS:GO on Mac runs like shit anyways I have a hard time making changes and testing it on windows.

**Comment by fluffyst** [comment]
I believe the culprit is [`BeamBoxAll`](https://github.com/nikooo777/ckSurf/blob/master/csgo/addons/sourcemod/scripting/ckSurf/surfzones.sp#L365) but I'm not 100% sure. I fixed it unintentionally but I don't have zones displaying to all clients at once.

**Comment by mloveless1227** [comment]
I have been testing this for the past couple of days and have yet to experience any of the fps problems I was having before. This is just an initial assessment but no issues reported and none I have personally experienced from the few hours of play I've had on it.
